tests/federation_room_send_test.go

```diff
@@ -249,3 +253,70 @@ func TestNetworkPartitionOrdering(t *testing.T) {
 		},
 	})
 }
+
+func TestAllowInvalidTransactionPDU(t *testing.T) {
+	deployment := complement.Deploy(t, 1)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Register(t, "hs1", helpers.RegistrationOpts{})
+
+	srv := federation.NewServer(t, deployment,
+		federation.HandleKeyRequests(),
+		federation.HandleMakeSendJoinRequests(),
+		federation.HandleTransactionRequests(nil, nil),
+	)
+
+	cancel := srv.Listen()
+	defer cancel()
+
+	nexy := srv.UserID("nexy")
+
+	room := alice.MustCreateRoom(t, map[string]interface{}{
+		"preset": "public_chat",
```

Comment:
As the log below wants room version 10, specify it, as it otherwise defaults to the server implementations default (which might be unknown).
```suggestion
		"preset": "public_chat",
		"room_version": "10",
```